### ISSUE TITLE
iOS: User's feed list is empty

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1765]
See this thread: https://forum.newsblur.com/t/newsblur-ios-app-empty-after-update-to-12-0-1-on-ios-16-1-1/9717/2

Username is dylanf. Try logging in and seeing if you can reproduce the issue.

**Comment by samuelclay** [comment]
I don't know that this will work as the user only has 1,840 feeds.

**Comment by Dejal** [comment]
It did work; I tested it with their account. The app was unresponsive with their feeds, when I limited it to 10k feeds, it was usable but sluggish, dropping the limit to 5k was much better.

As I said:

> Their “Main Folder” contains 80,744 feeds, which the table view can’t handle; it now limits display to the first 5,000 feeds in a folder.

Perhaps they've removed some excess feeds since then?

**Comment by Dejal** [comment]
I've investigated further. I see the issue. It was caused by changes from #1718 (nested folder bug):

> - A folder that has only subfolders, without any directly contained feeds, is now displayed in the list.
> - Selecting a folder will now show the feeds within all subfolders (plus any direct feeds).

In this case, the user has a "Main Folder" without any feeds, with subfolders with (it appears) the same feeds in multiple folders. So they are all added to the "Main Folder", resulting in the aforementioned 80k feeds in that folder.

I've modified the logic to de-dupe the feeds when adding to a parent folder, which solves the issue.
